### ISSUE TITLE
Labdir position for git repo sourced topology definition is wrong

### ISSUE THREAD

**Issue by steiler** [srl-labs/containerlab#1725]
@hellt you've introduced this chdir lately.
https://github.com/srl-labs/containerlab/blame/31220cffac886dd6d6a94b8027e3902b33eaea41/cmd/root.go#L178

This has the side effect, in the context of a git repository based deploy, that the cwd is changed to the working tree of the cloned repository. In my view at least for the git operation we should not change dir and keep the lab dir outside of the git repository working dir.

```
srl-telemetry-lab/
├── clab-st
├── configs
├── gnmic-config.yml
├── LICENSE
├── README.md
├── st.clab.yml
├── telemetry01.drawio
└── traffic.sh
```

Should be:
```
clab-st/
srl-telemetry-lab/
├── configs
├── gnmic-config.yml
├── LICENSE
├── README.md
├── st.clab.yml
├── telemetry01.drawio
└── traffic.sh
```

**Comment by hellt** [comment]
I don't think this is a valid ask.

Lab dir belongs to the same dir (by default) where clab file is located when you consider that using url to deoloy a lab is just a convenience function. It follows the same flow as cloning the repo, entering the cloned repo and calling deploy. 

It would be amazing if we could enter in the clones dir for a user, buy I think this is not possible. 

Some automation inside the lab may use the files present in the labdir (like exported topology json) and paths assumed to be relative to the git repo. 

Hope this makes sense

**Comment by steiler** [comment]
The issue I see is, that we leave an unclean worktree.
If you consider users changing branches via the `deploy -t <repo/branch>` we'd be changing with unclean workdir. Which might not be an issue thus far. But you could also make changes to the repo based files. IMHO it would make sense to not have the labdir as part of the repo, such that a `git add -a`  would not polute the repo.
This would overall allow the cloned repo to not be some sort of a throwaway instance.
Further your logic breaks when the topology file resides in a dir that's furhter down the tree in the repo. Also then the labdir would not be created next to the topo file but in the root of the repo.

Hope this also makes sense.

**Comment by hellt** [comment]
The repo should always have a gitignore for clab- pattern (labdir). This is present in all our labs

**Comment by hellt** [comment]
> Further your logic breaks when the topology file resides in a dir that's furhter down the tree in the repo. Also then the labdir would not be created next to the topo file but in the root of the repo.

I don't think so. My example was provided for a lab file at the root of the repo. When we deploy a file in a nested subdir we still work as a user would. Calling deploy from repo root providing a path to the file relative to repos root. 
Again, the labdir would be in the repo root, this behavior matches. 

With a gitignore file thay we always have in the labs the behavior for creating the labdir when using URL matches the behavior of a user working with the repo cloned manually
